File: go/space-age/space_age.go
package space

// Planet string type used for planet
type Planet string

// Age returns the number of years of age on planet given seconds
func Age(seconds float64, planet Planet) float64 {
	fp := func(sec2 float64) float64 {
		return Age(sec2, "Earth")
	}
	switch planet {
	case "Earth":
		return seconds / 31557600
	case "Mercury":
		return fp(seconds) / 0.2408467
	case "Venus":
		return fp(seconds) / 0.61519726
	case "Mars":
		return fp(seconds) / 1.8808158
	case "Jupiter":
		return fp(seconds) / 11.862615
	case "Saturn":
		return fp(seconds) / 29.447498
	case "Uranus":
		return fp(seconds) / 84.016846
	case "Neptune":
		return fp(seconds) / 164.79132
	default:
		return 0.0
	}
}

File: javascript/space-age/space-age.js
//
// This is only a SKELETON file for the 'Space Age' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const age = (planet, ageVal) => {
  let retVal
  switch(planet) {
    case "earth": retVal = earthAge(ageVal); break;
    case "mercury": retVal = earthAge(ageVal)/0.2408467; break;
    case "venus": retVal = earthAge(ageVal)/0.61519726; break;
    case "mars": retVal = earthAge(ageVal)/1.8808158; break;
    case "jupiter": retVal = earthAge(ageVal)/11.862615; break;
    case "saturn": retVal = earthAge(ageVal)/29.447498; break;
    case "uranus": retVal = earthAge(ageVal)/84.016846; break;
    case "neptune": retVal = earthAge(ageVal)/164.79132; break;
  }
  return strip(retVal);
};
const earthAge = (ageVal) => ageVal/31557600;

function strip(v) {
  return parseFloat(v.toFixed(2));
}
